### PR TITLE
Add missing translations and alphabetize yaml

### PR DESCRIPTION
**Pull request by AdrianaCeric**
**Description:**
Translations for various components of the app were missing, especially on the menu of the web app when viewed on mobile. Let me know if you notice errors and/or any other missing translations.

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

**Extra note**
I am not sure why the language selector in the menu is causing an error in the console when all the languages shown are translated/displayed correctly:
Pasted Graphic 22
image

### CODE REVIEW

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM with some minor changes.

File: i18n/es.yml [hunk 1]
```diff
@@ -116,11 +116,13 @@ components:
     mainTitle: Mis ajustes
     terms: Términos
   ItinerarySummary:
+    itineraryDetails: Detalles del itinerario
     minMaxFare: "{minTotalFare} - {maxTotalFare}"
   MapLayers:
     bike-rental: '{companies} Bicicleta Compartida'
     car-rental: Ubicaciones de los coches de alquiler
     micromobility-rental: '{companies} E-Scooters'
+    shared-vehicles: "Vehículos compartidos"
```
[the remark this answers is not in the document]
Comment (AdrianaCeric) [reply i18n/es.yml]: Fixed in my next commit

File: i18n/en-US.yml [hunk 2]
```diff
@@ -707,15 +712,15 @@ common:
     bus: Bus
     # The original OTP mode id is CABLE_CAR. Lowercase makes it cable_car.
     cable_car: Cable Car
-    car: Car
     car_park: Park and Ride
+    car: Car
```
Comment (philip-cline) [inline i18n/en-US.yml:716]: I think car should be sorted before car_park, but that's pretty nit picky.

File: i18n/en-US.yml [hunk 3]
```diff
@@ -707,15 +712,15 @@ common:
...
     drive: Drive
     ferry: Ferry
     flex: Flex Routes
     funicular: Funicular
     gondola: Gondola
-    micromobility: E-Scooter
     micromobility_rent: E-Scooter
+    micromobility: E-Scooter
```
Comment (philip-cline) [inline i18n/en-US.yml:723]: Same thing here for micromobility vs micromobility_rent

File: i18n/en-US.yml [hunk 4]
```diff
@@ -799,8 +804,8 @@ config:
     micromobility_rent: Rented E-scooter
   # Default values for Flex Indicator (set in configuration as well)
   flex:
-    flex-service: Flex Service
-    flex-service-colon: "Flex Service:"
     both: See bottom of itinerary for details
     call-ahead: Call to reserve
     continuous-dropoff: Communicate with operator about stop
+    flex-service-colon: "Flex Service:"
```
Comment (philip-cline) [inline i18n/en-US.yml:810]: Likewise

**Review by philip-cline — APPROVED** [review]
couple nits, but looks good to me.

File: i18n/en-US.yml (hunk 2 above)
Comment (AdrianaCeric) [reply i18n/en-US.yml:716]: I appreciate nit-picky-ness!

**Pull request merged**